sno/working_copy/sqlserver_adapter.py

```diff
@@ -65,12 +66,40 @@ def quote(ident):
 # geometry type info - the geometry type and CRS - is not roundtripped.
 
 
+# Used for constraining a column to be of a certain type, including subtypes of that type.
```

Comment:
So, PostGIS at least doesn't work this way. If I create a `GEOMETRYCOLLECTION` column that's the _only_ type I can insert into it. Which I think makes sense, because Simple Features for SQL doesn't have any "is-a" type functions beyond [`ST_GeometryType()`](https://postgis.net/docs/ST_GeometryType.html) so it's not easy to actually check whether you have a subtype.

But [GeoPackage explicitly allows assignable subtypes](https://www.geopackage.org/spec120/#r32). So I guess we go with that... (does that cause issues for PostGIS type constraints? Maybe we need some more tests around subtypes)

Reply:
And to be helpful Shapefiles mix MULTIX & X types freely in an "X" column, so datasets originating from there will potentially be a mix, or otherwise we cast to MULTIX. Thoughts @craigds

Reply:
Does that mean, if you create `GEOMETRY` column, then you can only put `GEOMETRY` types into it? That seems suspect...

Reply:
Geometry is an abstract type, so there's no check constraint and you can add anything into it.

Reply:
Yeah, I'm happy to go right now with:
- `GEOMETRY` → anything (no constraint)
- other types → them & their subtypes

Then revisit separately and figure out whether we tighten to the PG model or go with the GPKG approach in the general case.